python-sdk/tutorials/automl-with-azureml/image-classification-multilabel/auto-ml-image-classification-multilabel.ipynb

```diff
@@ -636,35 +684,127 @@
+    "def base64_to_img(base64ImgString):\n",
+    "    base64Img = base64ImgString.encode('utf-8')\n",
+    "    decoded_img = base64.b64decode(base64Img)\n",
+    "    return BytesIO(decoded_img).getvalue()"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
     "import requests\n",
-    "from IPython.display import Image\n",
+    "# from IPython.display import Image\n",
     "\n",
     "# URL for the web service\n",
     "scoring_uri = aks_service.scoring_uri\n",
     "\n",
     "# If the service is authenticated, set the key or token\n",
     "key, _ = aks_service.get_keys()\n",
     "\n",
-    "sample_image = \"./test_image.jpg\"\n",
-    "\n",
-    "# Load image data\n",
-    "data = open(sample_image, \"rb\").read()\n",
-    "\n",
     "# Set the content type\n",
     "headers = {\"Content-Type\": \"application/octet-stream\"}\n",
     "\n",
     "# If authentication is enabled, set the authorization header\n",
-    "headers[\"Authorization\"] = f\"Bearer {key}\"\n",
+    "headers[\"Authorization\"] = f\"Bearer {key}\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Scoring"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Inference on single image in bytes format"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_image = \"./test_image_1.jpg\"\n",
+    "\n",
+    "# Load image data\n",
+    "data = open(sample_image, \"rb\").read()\n",
     "\n",
     "# Make the request and display the response\n",
     "resp = requests.post(scoring_uri, data, headers=headers)\n",
-    "print(resp.text)"
+    "predictions = json.loads(resp.text)\n",
+    "predictions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Inference on single image in base64 format"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_image = \"./test_image_1.jpg\"\n",
+    "json_data = json.dumps({'image': img_to_base64(sample_image)})\n",
+    "resp = requests.post(scoring_uri, json_data, headers=headers)\n",
+    "predictions = json.loads(resp.text)\n",
+    "predictions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Inference on multiple images in base64 format"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_images = [\"./test_image_1.jpg\", \"./test_image_2.jpg\"]\n",
+    "json_data = json.dumps(\n",
+    "    {\"image\": [img_to_base64(img_path) for img_path in sample_images]}\n",
+    ")\n",
+    "resp = requests.post(scoring_uri, json_data, headers=headers)\n",
+    "predictions = json.loads(\"[{}]\".format(resp.text.replace(\"}{\", \"},{\")))\n",
```

Comment:
what prevents just json.loads(resp.text)?

Reply:
if the server returns in JSONL, should it return JSON? there's probably some awesome reasoning / great context that I'm missing

Reply:
nit: if we have to return JSONL, maybe we could have a util method or comment that converting from JSONL to JSON or something. maybe not -- just thinking out loud

Reply:
removing batch support so irrelevant now. but I was doing that because json.loads fails to parse the output. WIll consider your recommendations if we use it later